contracts/MockTrigger.sol

```diff
@@ -1,26 +1,55 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.4;
 
-import "./TriggerInterface.sol";
+import "./ITrigger.sol";
+
+contract MockTrigger is ITrigger {
+  /// @notice If true, checkAndToggleTrigger will toggle the trigger on its next call
+  bool public shouldToggle;
 
-contract MockTrigger is TriggerInterface {
   constructor(
     string memory _name,
     string memory _symbol,
     string memory _description,
     uint256[] memory _platformIds,
-    address _recipient
-  ) TriggerInterface(_name, _symbol, _description, _platformIds, _recipient) {}
+    address _recipient,
+    bool _shouldToggle
+  ) ITrigger(_name, _symbol, _description, _platformIds, _recipient) {
+    shouldToggle = _shouldToggle;
+
+    // Verify market is not already triggered.
+    require(!isMarketTriggered(), "Already triggered");
+  }
+
+  /**
+   * @notice Special function for this mock trigger to set whether or not the trigger should toggle
+   */
+  function setShouldToggle(bool _shouldToggle) external {
+    shouldToggle = _shouldToggle;
```

Comment:
Consider adding a `require(!isTriggered)` with a useful error message as a convenience to the developer. If this method is ever called after `isTriggered`, the dev probably messed something up and it's better to let them know.

Reply:
Done in 004ecce97745bd204e600454e08f32c7351527d0